crates/storage/nippy-jar/src/cursor.rs

```diff
@@ -0,0 +1,146 @@
+use crate::{compression::Compression, Filter, NippyJar, NippyJarError, PerfectHashingFunction};
+use memmap2::Mmap;
+use std::{clone::Clone, fs::File, sync::Mutex};
+use sucds::int_vectors::Access;
+use zstd::bulk::Decompressor;
+
+/// Simple cursor implementation to retrieve data from [`NippyJar`].
+pub struct NippyJarCursor<'a> {
+    /// [`NippyJar`] which holds most of the required configuration to read from the file.
+    jar: &'a NippyJar,
+    /// Optional dictionary decompressors.
+    zstd_decompressors: Option<Mutex<Vec<Decompressor<'a>>>>,
```

Comment:
why do we need a mutex here?

Reply:
decompressor needs to be mutable to decompress... which is less than ideal for sharing it across places. this is temporary, and will be changed in the future. (there's a missing Arc)

Reply:
makes sense, but we already take `&mut self` when using decompressor, so we should be good?